scripts/core/editor3/reducers/toolbar.tsx

```diff
@@ -115,6 +125,81 @@ const applyLink = (state, {link, entity}) => {
     return onChange(state, editorState);
 };
 
+interface IBlockRange {
+    offset: number;
+    length: number;
+    key: number;
+}
+
+/**
+ * Applies the given URL to the current content selection in multi-line quote block.
+ * If the selection is a link, it applies the link to the entity instead.
+ */
+const applyLinkToMultiLineQuote = (state, {link, entity}: {link: any, entity: EntityInstance}) => {
+    const {activeCell, editorState: mainEditorState} = state;
+
+    if (activeCell === null) {
+        return state;
+    }
+
+    const {i, j, key, currentStyle, selection} = activeCell;
+    const contentState = mainEditorState.getCurrentContent();
+    const block = contentState.getBlockForKey(key);
+    const data = getData(contentState, block.getKey());
+    const cellEditorState = getCell(data, i, j, currentStyle, selection);
+    let cellEditorStateData: EditorState;
+
+    // Check if the selection is a link
+    const selectedBlockKey = cellEditorState.getSelection().getAnchorKey();
+    const blockRanges: Array<IBlockRange> = entity.getData().data.cells[0][0].blocks
```

Comment:
How can I type a function which is coming from `draft-js`? If you mean to type the variable so that we have the full `data` structure typed I don't see the need for that in this use case since we'll never have a use for it - if you check one line down I want only the `entityRanges` and if you check line 157 you'll also see I use only `blockRanges`.

Reply:
add type, deserialize